### PR TITLE
feat: allow the use of management groups to assign permissions

### PR DESCRIPTION
**Pull request by alannix-lw**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -21,7 +21,7 @@ locals {
 data "azurerm_subscription" "primary" {}
 resource "azuread_application" "lacework" {
   count           = var.create ? 1 : 0
-  name            = var.application_name
+  display_name    = var.application_name
```
Comment (afiune) [inline main.tf:24]: is this a breaking change? or actually, fixing the old deprecated argument we have been seeing?

File: main.tf [hunk 2]
```diff
@@ -109,15 +109,21 @@ resource "azurerm_role_assignment" "grant_reader_role_to_subscriptions" {
...
 
-resource "random_password" "generator" {
-  count  = var.create ? 1 : 0
-  length = var.password_length
-}
-
 resource "azuread_application_password" "client_secret" {
   count                 = var.create ? 1 : 0
   application_object_id = azuread_application.lacework[count.index].object_id
-  value                 = random_password.generator[count.index].result
```
Comment (afiune) [inline main.tf:120]: we don't need this anymore?

**Review by afiune — COMMENTED** [review]
I do need to understand why are we not using the password generator?

**Comment by marcosgm** [comment]
Regarding Management Group reader permissions: it's 99% OK, it works, it's just that  we may want to do some housekeeping in a future PR. The variables "subscriptions" and "all_subscriptions" when combined with "use_management_group" can give a result that's not completely clean (but safe), i.e. double granting Reader permissions on a subscription and on its "parent" management group. Not a big deal, though.

File: main.tf (hunk 1 above)
Comment (alannix-lw) [reply main.tf:24]: This simply fixes the deprecation warning we see when running an `apply` - before/after show no pending changes from my testing.

File: main.tf (hunk 2 above)
Comment (alannix-lw) [reply main.tf:120]:
```
The value field will become read-only as Azure Active Directory no longer accepts user-supplied password values. Passwords will instead be auto-generated by Azure and will be exported as attributes by the resource.
```
https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/guides/microsoft-graph#resource-azuread_application_password

**Review by afiune — APPROVED** [review]
Amazing work, Alan!

tenor-127114963

**Pull request merged**